File: template_rest_assured_java/templates/src/test/java/br/com/back/tests/CreateUserTests.java
package br.com.back.tests;

import com.github.javafaker.Faker;
import io.restassured.response.ValidatableResponse;
import org.junit.Test;

import static br.com.back.service.UserService.*;
import static br.com.back.support.utils.BaseRest.*;
import static br.com.back.support.utils.GetVariable.getProp;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class CreateUserTests {
    ValidatableResponse response;

    static Faker faker = new Faker();
    static String name = faker.name().firstName();
    static String email = faker.internet().emailAddress();

    @Test
    public void createUserWithSuccess(){
        createUser(name, email, "123456","false");

        response = getResponse().then().log().all();

        assertEquals(201,response.extract().statusCode());
        assertEquals(getProp("message"), response.extract().path("message"));
        assertNotNull(response.extract().path("_id"));

    }
}

File: template_rest_assured_kotlin/templates/src/test/kotlin/br/com/back/tests/CreateUserTests.kt
package br.com.back.tests

import br.com.back.service.UserService.createUser
import io.restassured.response.ValidatableResponse
import com.github.javafaker.Faker
import org.junit.Assert
import org.junit.Test
import br.com.back.support.utils.BaseRest.response
import br.com.back.support.utils.GetVariable.getProp

class CreateUserTests {
    var responseRequest: ValidatableResponse? = null
    @Test
    fun createUserWithSuccess() {
        createUser(name, email, "123456", "false")
        responseRequest = response!!.then().log().all()
        Assert.assertEquals(201, responseRequest!!.extract().statusCode().toLong())
        Assert.assertEquals(getProp("message"), responseRequest!!.extract().path("message"))
        Assert.assertNotNull(responseRequest!!.extract().path("_id"))
    }

    companion object {
        var faker = Faker()
        var name = faker.name().firstName()
        var email = faker.internet().emailAddress()
    }
}